src/triage/component/architect/label_generators.py

```diff
@@ -21,29 +22,33 @@ def clean_up(self, labels_table_name):
 
 
 class LabelGenerator(object):
-    def __init__(self, db_engine, query, label_name=None):
+    def __init__(self, db_engine, query, label_name=None, replace=True):
         self.db_engine = db_engine
-
+        self.replace = replace
         # query is expected to select a number of entity ids
         # and an outcome for each given an as-of-date
         self.query = query
         self.label_name = label_name or DEFAULT_LABEL_NAME
 
     def _create_labels_table(self, labels_table_name):
-        self.db_engine.execute("drop table if exists {}".format(labels_table_name))
-        self.db_engine.execute(
-            """
-            create table {} (
-            entity_id int,
-            as_of_date date,
-            label_timespan interval,
-            label_name varchar(30),
-            label_type varchar(30),
-            label int
-        )""".format(
-                labels_table_name
+        if self.replace or not table_exists(labels_table_name, self.db_engine):
+            self.db_engine.execute("drop table if exists {}".format(labels_table_name))
+            self.db_engine.execute(
+                """
+                create table {} (
```

Comment:
Would it be a good idea to quote the table name? (I don't know the deep-down of how it's generated, but that's generally a good idea.)

Easy enough to "quote" here and above.

Reply:
Reading this Stackoverflow makes me not want to: https://stackoverflow.com/questions/6331504/omitting-the-double-quote-to-do-query-on-postgresql

Reply:
That's terrible ... PostgreSQL.

Casing fun aside, is there not any concern that the table name would have weird characters in it? If not, then yeah, no big deal.

Reply:
technically the LabelGenerator isn't enforcing anything about the table name, but in the Experiment the way it's constructed makes this not an issue.